From 326e4d70bc6033c4e13caae3a79f9cd637b2226d Mon Sep 17 00:00:00 2001
From: Jacek Danecki <Jacek.Danecki@intel.com>
Date: Tue, 14 Jan 2014 07:15:20 +0800
Subject: [PATCH 070/143] drm/i915: add registers needed by mfx_transcoder_drm

MFC_BITSTREAM_BYTECOUNT_FRAME - Reported Bitstream Output Byte Count per Frame Register
MFC_BITSTREAM_SE_BITCOUNT_FRAME - Reported Bitstream Output Bit Count for Syntax Elements Only Register
MFC_IMAGE_STATUS_MASK - MFC Image Status Mask
MFC_IMAGE_STATUS_CONTROL - MFC Image Status Control
---
 drivers/gpu/drm/i915/i915_cmd_parser.c |    4 ++++
 drivers/gpu/drm/i915/i915_reg.h        |    5 +++++
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_cmd_parser.c b/drivers/gpu/drm/i915/i915_cmd_parser.c
index 2f85bb7..98951b5 100644
--- a/drivers/gpu/drm/i915/i915_cmd_parser.c
+++ b/drivers/gpu/drm/i915/i915_cmd_parser.c
@@ -511,6 +511,10 @@ static const struct drm_i915_reg_descriptor hsw_master_regs[] = {
 
 static const struct drm_i915_reg_descriptor gen7_vcs_regs[] = {
 	REG32(MFD_ERROR_STATUS),
+    REG32(MFC_BITSTREAM_BYTECOUNT_FRAME),
+    REG32(MFC_BITSTREAM_SE_BITCOUNT_FRAME),
+    REG32(MFC_IMAGE_STATUS_MASK),
+    REG32(MFC_IMAGE_STATUS_CONTROL),
 };
 
 #undef REG64
diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 9f5d6c0..3f2875a 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -8154,4 +8154,9 @@ enum skl_disp_power_wells {
 #define GEN9_BLT_MOCS_0		0xcc00	/* Blitter MOCS base register*/
 
 #define MFD_ERROR_STATUS 0x12400
+#define MFC_BITSTREAM_BYTECOUNT_FRAME 0x124A0
+#define MFC_BITSTREAM_SE_BITCOUNT_FRAME 0x124A4
+#define MFC_IMAGE_STATUS_MASK 0x124B4
+#define MFC_IMAGE_STATUS_CONTROL 0x124B8
+
 #endif /* _I915_REG_H_ */
-- 
1.7.1

